tests/model_hub_tests/transformation_tests/test_stateful_to_stateless_transformation.py

```diff
@@ -8,21 +8,46 @@
 import models_hub_common.utils as utils
 import pytest
 import os
+import re
 
 def get_read_value_ops(model: ov.Model):
     return [op for op in model.get_ops() if op.get_type_name() == 'ReadValue']
 
-def check_desc_tensors(tensors1, tensors2):
-    # order of tensors may not match, comparing by the total amount and names
-    assert len(tensors1) == len(tensors2)
-    assert set(tuple(t.names) for t in tensors1) == set(tuple(t.names) for t in tensors2)
-    for t1 in tensors1:
-        t2_candidates = [t for t in tensors2 if t1.names & t.names]
-        assert len(t2_candidates) == 1
-        t2 = t2_candidates[0]
-        assert t1.names == t2.names
-        assert t1.get_partial_shape() == t2.get_partial_shape()
-        assert t1.get_element_type() == t2.get_element_type()
+def check_desc_tensors(expected_tensors, tensors):
+    # checks if tensors descriptors are same as expected
+    pattern_input = re.compile(R"input_restored.((past_key_values|present)\.(\d+)\.(key|value))")
+
+    assert len(expected_tensors) == len(tensors)
+    for expected in expected_tensors:
+        # The `patch_stateful` in optimum use any name instead found key/value names OV will use names restore path
+        # Restore expected names to find tensor for compare (can be removed when HG optimum updated)
+        expected_names = {m[1] if m else name for m, name in ((pattern_input.match(name), name) for name in expected.names)}
+        # tensor names check is relaxed the expected is sub-set of final names
+        t_candidates = [t for t in tensors if expected_names.issubset(t.names)]
+        assert len(t_candidates) == 1
+        tensor = t_candidates[0]
+        assert expected.get_element_type() == tensor.get_element_type()
+        assert expected.get_partial_shape() == tensor.get_partial_shape()
+
+
+def check_result_desc_tensors(expected_tensors, tensors):
+    # checks if Result tensors descriptors are same as expected
+    pattern_restore_output = re.compile(R"output_restored.((past_key_values|present)\.(\d+)\.(key|value))")
+    pattern_output = re.compile(R"(present\.(\d+)\.(key|value))")
+
+    assert len(expected_tensors) == len(tensors)
+    for expected in expected_tensors:
+        # The `patch_stateful` in optimum use any name instead found key/value names OV will use names restore path
+        # Restore expected names to find tensor for compare (can be removed when HG optimum updated)
```

Comment:
What is the problem with HG optimum? Does this mean that get_any_name returns different names now?

Reply:
The `get_any_name` is not reliable to use for testing to compare names when tensor can have many names or when some names are added to tensor.

As Result now not set its input names (they are not lost) the additional names can appear on compiled/transformed model. If there is many names get any name can give different name than on origin model and test fail but origin name is still on tensor.

This modification will only take stateless names on inputs and outputs and check if they are still present on model after transformation.

The HG issue was that it use any name and as there could be more names on tensor it pick the stateless pattern name or other. In case other name has been used the restore name path in transformation is used. This change will give more robust check whatever path is selected in transformation it will extract the names and prepare expected  and check if exists on input/outputs  on model after transformation.

The HG issue with any name has been fixed but OV can still use older version.